solidity/contracts/ManagedGrantFactory.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.5.4;
+
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+
+import "./utils/BytesLib.sol";
+
+import "./KeepToken.sol";
+import "./ManagedGrant.sol";
+import "./TokenGrant.sol";
+import "./GrantStakingPolicy.sol";
+
+contract ManagedGrantFactory {
+    using SafeERC20 for KeepToken;
+    using BytesLib for bytes;
+    using BytesLib for address;
+
+    KeepToken public token;
+    mapping(address => uint256) public grantFundingPool;
+    TokenGrant public tokenGrant;
+    GrantStakingPolicy nonRevocableStakingPolicy;
+    GrantStakingPolicy revocableStakingPolicy;
+
+    constructor(
+        address _tokenAddress,
+        address _tokenGrant,
+        address _nonRevocableStakingPolicy,
+        address _revocableStakingPolicy
+    ) public {
+        token = KeepToken(_tokenAddress);
+        tokenGrant = TokenGrant(_tokenGrant);
+        nonRevocableStakingPolicy = GrantStakingPolicy(_nonRevocableStakingPolicy);
+        revocableStakingPolicy = GrantStakingPolicy(_revocableStakingPolicy);
+    }
+
+    function receiveApproval(
+        address _from,
+        uint256 _amount,
+        address _token,
+        bytes memory _extraData
+    ) public {
+        require(KeepToken(_token) == token, "Invalid token contract");
+        require(_amount <= token.balanceOf(_from), "Insufficient sender balance");
```

Comment:
The balance check can be removed: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v2.4.0/contracts/token/ERC20/ERC20.sol#L158

We can also eliminate it from `TokenGrant.sol`. This was noted during the audit and somehow we missed to address it.